src/electron/platform/android/setup/steps/detect-permissions.ts

```diff
@@ -6,7 +6,18 @@ import { AndroidSetupStepConfig } from 'electron/platform/android/setup/android-
 export const detectPermissions: AndroidSetupStepConfig = deps => ({
     actions: {},
     onEnter: async () => {
+        deps.setApplicationName(); // init
+
         const detected = await deps.hasExpectedPermissions();
-        deps.stepTransition(detected ? 'configuring-port-forwarding' : 'prompt-grant-permissions');
+
```

Comment:
The deviceFetcher actually makes an HTTP call to the service on port 62442, which means that it needs to occur _after_ the port forwarding has been handled. As such, this is in the wrong step. I'd suggest putting it into the configuring-port-forwarding step.